yara-validator/yara_validator.py

```diff
@@ -1187,54 +1205,205 @@ def __parse_scheme(self, cfg_to_parse):
+        :return: void
+        """
+        argument = params.get("argument")
+        if argument:
+            if argument.get("parent"):
+                self.required_fields[tag + place_holder] = self.required_fields.pop(tag)
+                tags_in_child_parent_relationship.append(argument.get("parent"))
+            elif argument.get("child"):
+                child_tag = argument["child"]
+                argument.update({"child_place_holder": child_tag + place_holder})
+                tags_in_child_parent_relationship.append(argument.get("child"))
+
+    def validate_child_parent_tags(self, configuration, tags_in_child_parent_relationship):
+        """
+        Checks if any tags in child-parent relationships are missing from CCCS_Yara.yml configuration page
+        :param configuration: CCCS_Yara.yml configuration in dictionary format
+        :param tags_in_child_parent_relationship: a list of tags in child-parent relationships
+        :return: void
+        """
+        for tag in tags_in_child_parent_relationship:
+            if configuration.get(tag) is None:
+                print("CCCS_Yara.yml: \"" + tag + "\" is required (in a child-parent relationship)")
+                exit(1)
+
+    def read_regex_values(self, file_name, regex_tag):
+        """
+        Parses multiple values under the name "regex_tag" from given YAML file to make a single line of expression
+        :param file_name: name of the file to reference
+        :param regex_tag: name of the tag in the file that contains multiple regex expressions
+        :return: single line of regex expression
+        """
+        regex_yaml_path = SCRIPT_LOCATION.parent / file_name
+        with open(regex_yaml_path, "r") as yaml_file:
+            scheme = yaml.safe_load(yaml_file)
+
+        cfg_being_parsed = ""
+        for index, cfg in enumerate(scheme[regex_tag]):
+            if index > 0:
+                cfg_being_parsed = cfg_being_parsed + "|"
+
+            cfg_being_parsed = cfg_being_parsed + "^" + str(cfg['value']) + "$"
+
+        return cfg_being_parsed
+
+    def read_yara_cfg(self, tag, params, tag_position):
+        """
+        Creates a TagAttributes object for self.required_fields based on the CCCS_Yara.yml configuration
+        :param tag: string name of a tag in CCCS_Yara.yml file
+        :param params: parameters of the corresponding metadata tag in dictionary format
+        :param tag_position: index (position) of the key in CCCS_Yara.yml file
+        :return: TagAttributes instance
+        """
+        # parameters for creating a TagAttributes instance
+        tag_max_count = None
+        tag_optional = None
+        tag_validator = self.valid_regex  # default validator if not provided in the configuration
+        tag_argument = {"regexExpression": UNVALIDATED_REGEX}  # default argument (regex expression) for default validator

```

Comment:
I don't think we want to default to a wildcard regex. We should fail the validation.